ejb3/src/main/java/org/jboss/as/ejb3/timerservice/TimerServiceImpl.java

```diff
@@ -1133,12 +1128,23 @@ public TimerServiceResource getResource() {
         return resource;
     }
 
-    private void registerTimerResource(final String timerId) {
-        this.resource.timerCreated(timerId);
+    private boolean registerTimerResource(final TimerImpl timer) {
+        synchronized (this.timers) {
+            if (this.timers.containsKey(timer.getId())) {
+                this.timers.put(timer.getId(), timer);
```

Comment:
If the timer id already exists in the internal cache `timers`, why do we put the timer to `timers` again? It looks harmless though.

Reply:
ok. removed